addon/lib/shooter.js

```diff
@@ -296,14 +296,33 @@ const ShotContext = Class({
       }
       this.updateShot();
     },
-    copyLink: function () {
-      clipboard.set(this.shot.viewUrl, "text");
+    copyLink: function (url) {
+      clipboard.set(url, "text");
       notifications.notify({
         title: "Link Copied",
         text: "The link to your shot has been copied to the clipboard.",
         iconURL: self.data.url("../data/copy.png")
       });
     },
+    copyImage: function (img) {
+      clipboard.set(img, "image");
+      notifications.notify({
+        title: "Image Copied",
+        text: "Your shot has been copied to the clipboard.",
+        iconURL: self.data.url("../data/copy.png")
+      });
+    },
+    copyRich: function (url, img, title) {
+      // Use "text" instead of "html" so that pasting into a text area or text editor
+      // pastes the html instead of the plain text stripped out of the html.
+      clipboard.set(
+        `<a href="${url}"><img src="${img}" /><div>${title}</div></a>`, "text");
```

Comment:
There's no good function.  You have to do `text.replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/"/g, "&quot;")` (or better put it in a function).

Also do we really want "text"?  We need to test a few scenarios:

- Pasting into Gmail (maybe also Yahoo mail or something)
- Pasting into some WYSIWYG, e.g., on WordPress (FCKEdit I think?)
- Pasting into Twitter and Facebook